crates/core/src/factories/core_fetch/test.rs

```diff
@@ -0,0 +1,600 @@
+use std::{
+    sync::{Arc, Mutex},
+    time::Duration,
+};
+
+use bytes::Bytes;
+use kitsune2_api::{
+    fetch::{deserialize_op_ids, Fetch},
+    id::Id,
+    transport::Transport,
+    AgentId, K2Error, OpId, SpaceId, Url,
+};
+use rand::Rng;
+
+use crate::{default_builder, factories::test_utils::AgentBuilder};
+
+use super::{CoreFetch, CoreFetchConfig};
+
+#[derive(Debug)]
+pub struct MockTransport {
+    requests_sent: Arc<Mutex<Vec<(OpId, Url)>>>,
+    fail: bool,
+}
+
+impl MockTransport {
+    fn new(fail: bool) -> Arc<MockTransport> {
+        Arc::new(Self {
+            requests_sent: Arc::new(Mutex::new(Vec::new())),
+            fail,
+        })
+    }
+}
+
+impl Transport for MockTransport {
```

Comment:
Ah, by the way I tried `mockall` for this, but I still had to write the `unimplemented` fns.